### PR TITLE
OUTDATED Upkeep becomes fixed, I believe + iron door nerf

### PR DESCRIPTION
**Pull request by Garodil**
whoops sorry one commit for two different changes

### CODE REVIEW

**Comment by Garodil** [comment]
If that pull request is accepted, every faction would have 1 slot for player. Every camp gives 1 slot, so 2 slots for every new faction. upgrade gives +1 slot for player. Coal mines takes one slot, witch shop gives 1 slot for players, merchant gives 1 slot. Quarters wouldn't give any slots. Slaves are less dangerous to upkeep too. Every player always take only 1 slot.

Iron door would have less hp(9 -> 7), also bullets do more damage to them. They fall apart if aren't attached to blocks, like stone doors. Stone block costs 5 stone - Stone door costs 50 stone. Iron block costs 2 iron ingots - with my pull request door would cost 8 iron ingots, instead of 4.

Changes CommonGun.as to make doors less resistant to bullets.

Completed. In theory, makes faction gameplay more attractive, also removes few annoying problems.

File: TFlippy_TerritoryControl_Structures_Dev/Entities/Structures/BombShop/BombShop.as [hunk 1]
```diff
@@ -11,9 +11,9 @@ void onInit(CBlob@ this)
 {
 	this.set_TileType("background tile", CMap::tile_castle_back);
 
-	this.Tag("upkeep building");
-	this.set_u8("upkeep cap increase", 0);
-	this.set_u8("upkeep cost", 5);
+//	this.Tag("upkeep building");
+//	this.set_u8("upkeep cap increase", 0);
+//	this.set_u8("upkeep cost", 5);
```
Comment (Vam-Jam) [inline TFlippy_TerritoryControl_Structures_Dev/Entities/Structures/BombShop/BombShop.as]: indent

File: TFlippy_TerritoryControl_Structures_Dev/Entities/Structures/ChickenMarket/ChickenMarket.as [hunk 2]
```diff
@@ -11,9 +11,6 @@
 
 void onInit(CBlob@ this)
 {
-	this.Tag("upkeep building");
-	this.set_u8("upkeep cap increase", 0);
-	this.set_u8("upkeep cost", 50);
```
Comment (Vam-Jam) [inline TFlippy_TerritoryControl_Structures_Dev/Entities/Structures/ChickenMarket/ChickenMarket.as:16]: could have commented this out like the other one

File: TFlippy_TerritoryControl_Patreon_Dev/Structures/PatreonShop/PatreonShop.as [hunk 3]
```diff
@@ -19,10 +19,6 @@ const SColor[] colors =
 void onInit(CBlob@ this)
 {
 	this.set_TileType("background tile", CMap::tile_castle_back);
-
-	this.Tag("upkeep building");
-	this.set_u8("upkeep cap increase", 0);
-	this.set_u8("upkeep cost", 5);
```
Comment (Vam-Jam) [inline TFlippy_TerritoryControl_Patreon_Dev/Structures/PatreonShop/PatreonShop.as:25]: comment out instead of remove

File: TFlippy_TerritoryControl_Structures_Dev/Entities/Structures/Armory/Armory.as [hunk 4]
```diff
@@ -12,10 +12,6 @@ Random traderRandom(Time());
 void onInit(CBlob@ this)
 {
 	this.set_TileType("background tile", CMap::tile_castle_back);
-
-	this.Tag("upkeep building");
-	this.set_u8("upkeep cap increase", 0);
-	this.set_u8("upkeep cost", 5);
```
Comment (Vam-Jam) [inline TFlippy_TerritoryControl_Structures_Dev/Entities/Structures/Armory/Armory.as:18]: comment out instead of remove

File: TFlippy_TerritoryControl_Structures_Dev/Entities/Structures/Gunsmith/Gunsmith.as [hunk 5]
```diff
@@ -27,9 +27,9 @@ void onInit(CBlob@ this)
 {
 	this.set_TileType("background tile", CMap::tile_castle_back);
 
-	this.Tag("upkeep building");
-	this.set_u8("upkeep cap increase", 0);
-	this.set_u8("upkeep cost", 5);
+//	this.Tag("upkeep building");
+//	this.set_u8("upkeep cap increase", 0);
+//	this.set_u8("upkeep cost", 5);
```
Comment (Vam-Jam) [inline TFlippy_TerritoryControl_Structures_Dev/Entities/Structures/Gunsmith/Gunsmith.as]: indent

File: TFlippy_TerritoryControl_Structures_Dev/Entities/Structures/Quarters/Quarters.as [hunk 6]
```diff
@@ -67,10 +67,6 @@ void onInit(CBlob@ this)
 
 	this.Tag("change team on fort capture");
 	
-	this.Tag("upkeep building");
-	this.set_u8("upkeep cap increase", 10);
-	this.set_u8("upkeep cost", 0);
-
```
Comment (Vam-Jam) [inline TFlippy_TerritoryControl_Structures_Dev/Entities/Structures/Quarters/Quarters.as]: comment instead of remove

**Review by Vam-Jam — CHANGES_REQUESTED** [review]
some small feedback

**Comment by Garodil** [comment]
> some small feedback

Done. Marked all of them as resolved, if you don't mind.

**Comment by Garodil** [comment]
I believe it is not really clear why I even have requested that upkeep tweak.

To be clear - TC have got now not very good upkeep system, that makes hard to create faction at first, but doesn't even help with splitting up players to multiply factions, instead of one megafaction. Megafaction can just make big quarters skyscrapers that are cheap and are enough to never again remember about upkeep. Obviously, everyone joins the skyscraper megafaction, because it is easier than create own faction. So, current upkeep does its work absolutely wrong way - creating hard conditions for new factions and easy for megafactions. Also, upkeep doesn't promote faction gameplay, conquering shops causes more cons, than pros, for example.

My Upkeep makes(I believe) it easy to control faction's upkeep, because of disabled most of structures relation to upkeep calc, also quarters doesn't give any upkeep, so it is harder to forget about upkeep completely. Though, it creates interesting situation, because factions would be meant to capture shops to become strong. It promotes megafactions by some degree, of course, but mostly it would(perhaps) cause multiply factions to appear at start. Also! With that way, factions always would have in mind maximum of players that they could get in their faction, so even if megafaction has established, it would have to create multiply factions bases, for the purpose to get everyone join their megafaction. Maybe it is unbalanced, actually, but I need actual experience with a lot of players on in order to make it balanced.

tl;dr:
Current TC upkeep does nothing. PR's one upkeep would make it easier to get multiply player factions, in theory.

**Comment by Gingerbeard5773** [comment]
This Upkeep system quite resembles the one that I have been thinking about for a few months now, with the same emphasis on player count rather than buildings or by capturables. Here is my analysis on Garodil's system.

One- The removal of cost from buildings (tinker shop etc) has been overdue for too long. This upkeep cost of built structures makes no sense, as it costs upkeep to have these buildings when a neutral can build them without the same cost. This is also a particular reason why even players like me don't play in factions most of the time, cause I don't want to have to grind for upkeep in the beginning and spend the extra time and resources when it is unnecessary.

Two- Quarters not giving upkeep boost is an interesting change that my same system applies as well. The emphasis here is that players will upgrade their factions/ build more of them or capture some territory rather than spam quarters all over the place- **However**, this means that players may stack factions in close proximity to receive enough upkeep as if they were quarters. This is still better than quarters in my opinion as they are much more expensive to create. But, if this is an apparent problem, a good solution would be to give allied factions a no build radius like if they were enemy factions. My system applies this in emphasis that players will _naturally_ expand their territory in order to gain upkeep, and to slow the progression of power within the faction so games last longer.

Three- Costs for Territory may need to be tweaked with testing. These costs are the most speculative, and depending on how they cost is how players will pick and choose what to capture. Garodil's system uses +1 from witch and merchant, and -1 from coalmines. From observation, it is very likely players will try to capture as much territory as they can in order to fulfill upkeep as well as reap the benefits. It may be a good influence, as it could be an interesting gameplay factor within wars. This gives an opportunity for players to capture enemy territory and weaken enemy upkeep. It also proposes the idea that if a faction is too reliant on territory to fuel their faction, that it can be taken away and they can be cut off from more players joining. An interesting concept.

Upkeep was put in for a reason and it should be fixed not made irrelevant. There are plenty of ways to make upkeep fulfill its purpose but still be fun and engaging.
